atoMEC/numerov.py

```diff
@@ -579,9 +581,262 @@ def num_propagate(xgrid, v, l, E):
+
+#     while True:
+#         try:
+#             joblib_folder = "".join(
+#                 random.choices(string.ascii_uppercase + string.digits, k=30)
+#             )
+#             os.mkdir(joblib_folder)
+#             break
+#         except FileExistsError as e:
+#             print(e)
+
+#     # dump and load the large numpy arrays from file
+#     data_filename_memmap = os.path.join(joblib_folder, "data_memmap")
+#     dump(W, data_filename_memmap)
+#     W = load(data_filename_memmap, mmap_mode="r")
+
+#     with Parallel(n_jobs=config.numcores) as parallel:
+#         X = parallel(
+#             delayed(num_propagate)(q, xgrid, W, config.lmax) for q in range(pmax)
+#         )
+
+#     # remove the joblib arrays
+#     try:
+#         shutil.rmtree(joblib_folder)
+#     except:  # noqa
+#         print("Could not clean-up automatically.")
+
+#     for q in range(pmax):
+#         eigfuncs_flat[q] = X[q]
+
+#     eigfuncs_e = eigfuncs_flat.reshape(config.spindims, config.lmax, n_e_grid, N)
+
+#     return eigfuncs_e
+
+
+# def num_propagate_2(xgrid, v, e_arr):
+#     """
+#     Propagate the wfn manually for fixed energy with numerov scheme.
+
+#     Parameters
+#     ----------
+#     xgrid : ndarray
+#         the logarithmic grid
+#     v : ndarray
+#         KS potential array
+#     l : int
+#         angular momentum value
+#     E : float
+#         energy of the wavefunction
+
+#     Returns
+#     -------
+#     Psi_norm : ndarray
+#         normalized wavefunction
+#     """
+#     # define some initial grid parameters
+#     dx = xgrid[1] - xgrid[0]
+#     x0 = xgrid[0]
+#     h = (dx ** 2) / 12.0  # a parameter for the numerov integration
+#     N = np.size(xgrid)  # size of grid
+
+#     # 'Potential' for numerov integration
+
+#     # Initial conditions
+#     # 'Potential' for numerov integration
+#     W = np.zeros((N, len(e_arr), config.lmax, config.spindims))
+#     for i in range(N):
+#         for sp in range(config.spindims):
+#             for l in range(config.lmax):
+#                 W[i, :, l, sp] = (
+#                     -2 * np.exp(2.0 * xgrid[i]) * (v[sp, i] - e_arr) - (l + 0.5) ** 2
+#                 )
+
+#     # Initial conditions
+#     Psi = np.zeros((N, len(e_arr), config.lmax, config.spindims))  # initialize wfn
+#     for l in range(config.lmax):
+#         Psi[:, :, l] = np.exp((l + 0.5) * (x0 + dx))
+
+#     # Integration loop
+#     for i in range(2, N):
+#         Psi[i] = (
+#             2.0 * (1.0 - 5.0 * h * W[i - 1]) * Psi[i - 1]
+#             - (1.0 + h * W[i - 2]) * Psi[i - 2]
+#         ) / (1.0 + h * W[i])
+
+#     # normalize the wavefunction
+#     Psi = Psi.transpose()
+#     psi_sq = np.exp(-xgrid) * Psi ** 2  # convert from P_nl to X_nl and square
+#     integrand = 4.0 * np.pi * np.exp(3.0 * xgrid) * psi_sq
+#     norm = (np.trapz(integrand, x=xgrid, axis=-1)) ** (-0.5)
+#     Psi_norm = np.einsum("ijl,ijlk->ijlk", norm, Psi)
+
+#     return Psi_norm
```

Comment:
Mhh, I don't really like the idea  of having a bunch of block comments of old code in the code base. My suggestion would be to create a special commit with a clear message indicating the removal of deprecated code that may become useful again in future versions. Also what is not clear to me is where this code is coming from? It doesn't look like you are converting the code to a block comment, but adding commented code here :thinking:.

Reply:
I will do as you suggest with the commit message and remove the commented out code, I agree it isn't so nice. But I'm not sure what you mean about converting the code to a block comment vs adding commented code? I just commented out some old functions.

Reply:
Hmm might be because I was working on two different implementations at the same time, and copied a block of code from the other branch in case it would be useful... Anyway have done as suggested with deletion of commented code and an appropriate commit message